workflows/views.py

```diff
@@ -90,3 +94,27 @@ def get(
         if workflow is None:
             raise NotFound()
         return FileResponse(workflow.data.file)
+
+    def put(
+        self,
+        request: Request,
+        user_id: str,
+        workflow_id: str,
+        _format: Optional[str] = None,
+    ) -> Response:
+        user = authed_user(request)
+        if user.username != user_id:
```

Comment:
Not part of the newly added code in this PR, but `WorkflowDetail.get` crashes if we pass a string as `user_id` as the database has integers here. We need to find the id for an user-name first.

This works:
GET localhost:8000/api/1/workflows/bla

This does not work:
GET localhost:8000/api/admin/workflows/bla

Reply:
Yeah that's intentional but poorly named. I actually changed that already in the code I'm working on now.

Reply:
I'm surprised that doesn't work. It should and it seems to for me.

Reply:
It also doesn't crash if you give it a user_id of 1 because it compares the authed username against `"1"` which likely fails unless the username is actually `1`.

The naming is bad though, I actually had already changed it to `username` in the stuff I'm doing now to prevent confusion about what should be there.

Reply:
Sorry for marking more or less unrelated code here but the code I mean was not part of this PR. I mean the following piece:

```
class WorkflowDetail(APIView):
...
    def get(
        self,
        _request: Request,
        user_id: str,
        workflow_id: str,
        _format: Optional[str] = None,
    ) -> Response:
        workflow = Workflow.objects.filter(name=workflow_id, author=user_id).first()
...
```

`author`(_id) in the db-table `workflows_workflow` is an integer, so if pass a string in here it crashes with: 

```
ValueError: Field 'id' expected a number but got 'admin'.
```

One way to fix it:

```
user_id = User.objects.get(username=username)
```

Reply:
But I am not sure if this will work with LDAP when the users are not inside the `auth_user` table (at the beginning?). But this will be a problem for Martin and me for the future I guess.

Reply:
Ah I see. Yeah I ran into that myself and fixed it already so I'll just commit that.